iree/compiler/Dialect/Flow/Transforms/CreateBenchmarkFuncs.cpp

```diff
@@ -0,0 +1,147 @@
+        if (!dispatchEntryOp) continue;
+        auto execFuncOp = execOp.getInnerModule().lookupSymbol<FuncOp>(
+            dispatchEntryOp.function_ref());
+
+        // Create a funcOp to invoke the dispatch function.
+        std::string funcName = std::string(execFuncOp.getName()) + "_entry";
+        auto funcType =
+            builder.getFunctionType({}, execFuncOp.getType().getResults());
+        auto funcOp = builder.create<FuncOp>(loc, funcName, funcType);
+        funcOp.setAttr("iree.module.export", UnitAttr::get(&getContext()));
+        Block* block = funcOp.addEntryBlock();
+
+        // Build the body of the FuncOp.
+        OpBuilder::InsertionGuard gaurd(builder);
+        builder.setInsertionPoint(funcOp);
+        auto blockBuilder = OpBuilder(block, block->begin());
+        SmallVector<Value, 4> args;
+        for (auto inputType : execFuncOp.getType().getInputs()) {
+          args.push_back(
+              getDummyInput(builder, blockBuilder, inputType, moduleSymbols));
+        }
+
+        // TODO(hanchung): Use a real workload instead? We can probably
+        // calculate the workload from the results.
+        auto dummyWorkload = blockBuilder.create<ConstantIndexOp>(loc, 0);
+        auto dispatchOp = blockBuilder.create<DispatchOp>(
+            loc, dispatchEntryOp, dummyWorkload, funcType.getResults(), args);
+        blockBuilder.create<mlir::ReturnOp>(loc, dispatchOp.getResults());
+      }
+    }
+
+    // TODO(#3577): Move below part to a separate pass and use CallOp instead of
+    // clone the region. The CallOp is materialized in an easlier stage. We
+    // don't expect to see it at flow level.
+    for (auto funcOp : moduleOp.getOps<FuncOp>()) {
+      if (!funcOp.getAttr("iree.module.export")) {
+        continue;
+      }
+      if (funcOp.getNumArguments() == 0) {
+        continue;
+      }
+
+      loc = funcOp.getLoc();
+      auto funcType =
+          builder.getFunctionType({}, funcOp.getType().getResults());
+      std::string funcName = std::string(funcOp.getName()) + "_dummy_args";
+      auto newFuncOp = builder.create<FuncOp>(loc, funcName, funcType);
+      newFuncOp.setAttr("iree.module.export", builder.getUnitAttr());
+      Block* block = newFuncOp.addEntryBlock();
+
+      OpBuilder::InsertionGuard gaurd(builder);
+      builder.setInsertionPoint(newFuncOp);
+      auto blockBuilder = OpBuilder(block, block->begin());
+      BlockAndValueMapping mapping;
+      for (auto iter : llvm::enumerate(funcOp.getType().getInputs())) {
+        auto arg =
+            getDummyInput(builder, blockBuilder, iter.value(), moduleSymbols);
+        mapping.map(funcOp.getArgument(iter.index()), arg);
+      }
+      for (auto& op : funcOp.getRegion().begin()->getOperations()) {
+        blockBuilder.clone(op, mapping);
+      }
+
+      funcOp.removeAttr("iree.module.export");
+    }
+  }
+
+ private:
+  std::string getUniqueName(const SymbolTable& moduleSymbols) {
```

Comment:
The symbol table already uniques names, so seems like this shouldn't be necessary? https://github.com/llvm/llvm-project/blob/c23cb8b032eb926f1d9c96c1415d6efd305eb59a/mlir/include/mlir/IR/SymbolTable.h#L42

Reply:
good to know this, thanks!